rust/lance/src/index/append.rs

```diff
@@ -16,25 +16,31 @@ use std::sync::Arc;
 
 use lance_core::{format::Index as IndexMetadata, Error, Result};
 use log::info;
+use roaring::RoaringBitmap;
 use snafu::{location, Location};
 use uuid::Uuid;
 
-use crate::dataset::index::unindexed_fragments;
+use crate::dataset::index::{indexed_fragment_ids, unindexed_fragments};
 use crate::dataset::Dataset;
 use crate::index::vector::ivf::IVFIndex;
 
 use super::DatasetIndexInternalExt;
 
 /// Append new data to the index, without re-train.
+///
+/// Returns the UUID of the new index along with a vector of newly indexed fragment ids
 pub async fn append_index(
     dataset: Arc<Dataset>,
     old_index: &IndexMetadata,
-) -> Result<Option<Uuid>> {
+) -> Result<Option<(Uuid, RoaringBitmap)>> {
     let unindexed = unindexed_fragments(old_index, dataset.as_ref()).await?;
     if unindexed.is_empty() {
         return Ok(None);
     };
 
+    let mut frag_bitmap = indexed_fragment_ids(old_index, &dataset).await?;
```

Comment:
can we add a minimal repro test as well?

Reply:
I'm adding that in a separate PR

Reply:
Yes, @wjones127 pointed out that we can possibly encounter these corrupt databases in the wild now and so this fix still doesn't handle that scenario. He had a suggestion for changing how we test for deleted row ids when entire fragments are deleted that should cover it.  We will also add a new "corrupt dataset" example to our test matrix